tortoise_interpreter.py

```diff
@@ -1,51 +1,77 @@
+"""Implementation of tortoise-language."""
+
+
 def main_tortoise(program, print_to_screen=print):
+    """Main function that interprets a tortoise program."""
     program = sanitize_program(program)
-    COMMANDS = {'U': pen_up, 'D': pen_down, 'P': pen_colour, 'N': move_north, 'S': move_south, 'E': move_east, 'W': move_west}
-    command_to_parse = navigate_command_functions(COMMANDS, program)
+    commands = {
+        "U": pen_up,
+        "D": pen_down,
+        "P": pen_colour,
+        "N": move_north,
+        "S": move_south,
+        "E": move_east,
+        "W": move_west,
+    }
+    command_to_parse = navigate_command_functions(commands, program)
 
     for command in command_to_parse:
         print_to_screen(command)
 
 
 def sanitize_program(program):
+    """Get tortoise program ready for interpretation."""
     # If a command has two parts, the second element of the list will be an empty string.
     # Remove this element. Then remove all elements after the first empty string in each list.
     # These were the comments at the end of each command.
     for command in program:
         if len(command) > 1:
             command.pop(1)
             if len(command) > 2:
-                del command[command.index(' '):]
+                del command[command.index(" ") :]
     return program
 
 
-def navigate_command_functions(COMMANDS, program):
-    return [f'{i}. {COMMANDS[command[0]](command)}' for i, command in enumerate(program, 1)]
+def navigate_command_functions(commands, program):
+    """Number each command of the tortoise program and call the
```

Comment:
Ideally, you want your first line of the docstring to be a short summary, and the details can go in the body. This is a bit long for one line.